src/luarocks/core/sysdetect.lua

```diff
@@ -192,6 +192,11 @@ local function detect_elf_system(fd, hdr, sections)
             return "linux"
          end
       end
+
+      local uname_s = io.popen("uname -s"):read("*l")
```

Comment:
The goal of the sysdetect module is to implement OS and architecture detection without forking processes. There should be a way to detect the OS and architecture uniquely by verifying some fixed location in the filesystem or the identifying headers of well-known executables.